tensorflow_io/core/plugins/file_system_plugins.cc

```diff
@@ -25,8 +31,14 @@ void TF_InitPlugin(TF_FilesystemPluginInfo* info) {
   tensorflow::io::az::ProvideFilesystemSupportFor(&info->ops[0], "az");
   tensorflow::io::http::ProvideFilesystemSupportFor(&info->ops[1], "http");
   tensorflow::io::s3::ProvideFilesystemSupportFor(&info->ops[2], "s3e");
-  tensorflow::io::hdfs::ProvideFilesystemSupportFor(&info->ops[3], "hdfse");
-  tensorflow::io::hdfs::ProvideFilesystemSupportFor(&info->ops[4], "viewfse");
-  tensorflow::io::hdfs::ProvideFilesystemSupportFor(&info->ops[5], "hare");
+  if (!(enable_legacy == "true" || enable_legacy == "1")) {
```

Comment:
nit: it would be more intuitive like this:
if (enable_legacy == "true" || enable_legacy == "1") {
     /// legacy file system case
}
else {
   /// non-legacy case
}